src/engine/Service.cpp

```diff
@@ -179,25 +179,31 @@ ProtoResult Service::computeResultImpl([[maybe_unused]] bool requestLaziness) {
                          std::back_inserter(expVariableKeys),
                          [](const Variable& v) { return v.name().substr(1); });
 
-  // Set basic properties of the result table.
-  IdTable idTable{getResultWidth(), getExecutionContext()->getAllocator()};
-  LocalVocab localVocab{};
-
   auto body = ad_utility::LazyJsonParser::parse(std::move(response.body_),
                                                 {"results", "bindings"});
 
-  // Fill the result table using the `writeJsonResult` method below.
-  CALL_FIXED_SIZE(getResultWidth(), &Service::writeJsonResult, this,
-                  expVariableKeys, body, &idTable, &localVocab);
+  auto localVocabPtr = std::make_shared<LocalVocab>();
+  auto generator = computeResultLazily(expVariableKeys, std::move(body),
```

Comment:
Add a comment "Note:  the `body` generator also keeps the complete response connection alive, so we have no lifetime issue here (see `HttpRequest::send` for details)."

That is the one thing one is wondering here, and then it is clarified.